cmd/bedpeOverlap/bedpeOverlap.go

```diff
@@ -80,11 +113,39 @@ func SelectIsBed(bedSelectFile string, bedpeInFile string, overlapThreshold floa
 			currOverlaps = interval.Query(selectTree, i.B, "any")
 			if len(currOverlaps) > 0 {
 				if overlapThreshold == 0 {
+					if keepNames {
+						i.A.FieldsInitialized = 7
+						for c := range currOverlaps {
+							if c == 0 {
+								i.A.Name = currOverlaps[c].(bed.Bed).Name
+							} else {
+								if i.A.Name == currOverlaps[c].(bed.Bed).Name {
+									continue
+								} else {
+									i.A.Name += "," + currOverlaps[c].(bed.Bed).Name
+								}
+							}
+						}
+					}
 					bedpe.WriteToFileHandle(out, i)
 				} else {
 					found = false
 					for _, j := range currOverlaps {
 						if !found && overlapPercent(j, i.B) >= overlapThreshold {
+							if keepNames {
+								i.A.FieldsInitialized = 7
```

Comment:
make sure a and b have at least 7 and if not then initialize 7, get rid of i and rename i currBedpe